packages/core/src/common/command.ts

```diff
@@ -36,6 +36,10 @@ export interface Command {
      * An icon class of this command.
      */
     iconClass?: string;
+    /**
+     * A prefix of this command.
+     */
+    prefix?: string;
```

Comment:
Do we need it? How about just adding a category for commands which does not have it yet?

Also, It should be called `category` not a prefix if we introduce it.

Reply:
Also prefixing does not make sense always, see discussions here: https://github.com/theia-ide/theia/issues/1119#issuecomment-372812915

Reply:
> Do we need it? How about just adding a category to command labels which don't have it yet?

I tried simply adding the category to the command labels but this had the unwanted side-effect of also affecting the menu item labels. The easiest way was for me to add a new field, and only in the display of the `quick-command-palette` would I prefix the commands with their category.

> Also, It should be called `category` not a prefix if we introduce it.

Sure, I'll make the change :)

Reply:
> Also prefixing does not make sense always, see discussions here: [#1119 (comment)](https://github.com/theia-ide/theia/issues/1119#issuecomment-372812915)

I tried to base myself as much as I could on how `vscode` prefixes their commands. For instance, if we prefix all `Terminal` commands, it makes it a lot easier in the future to add more Terminal commands that a user can easily search by. However, we can go on a case-by-case basis and determine what makes sense and what doesn't. I'll make another passthrough all the commands.